jamaljsr/polar
# feat(network): add support for c-lightning v0.7.3

### Description

This PR adds support for c-lightning v0.7.3 nodes. All of the existing functionality for LND nodes has been implemented for c-lightning as well (ex: open channels, launch terminal, etc). 

**Note:** c-lightning syncs chain data with the bitcoind node once every 30 secs. So most actions are not reflected immediately in the UI. Polar needs to wait 30s then query the node for updated channels & balances.

There is still a bit more work I have to finish before merging. This early PR is for anyone interested in c-lightning support to test and provide feedback.   

Remaining Improvements:
- [x] fix some c-lightning channels not being displayed
- [x] try to improve the UX around the c-lightning 30s chain sync delay
- [x] add and update unit tests
- [x] test full functionality on Windows & Linux

### Steps to Test

1. Create a network with 2x c-lightning and 1x LND nodes
2. Deposit funds and confirm balance updates correctly
3. Launch the Terminal and run a few lightning-cli commands
4. Open a channel to a c-lightning and a LND node
5. Close one of the channels
6. Remove the node from the network
7. Drag a new c-lightning node from the sidebar
8. Repeat steps 2-5

### Screenshots

![image](https://user-images.githubusercontent.com/1356600/69745771-dfcb1a00-1110-11ea-9899-ed220709f9a9.png)
